src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/sdl_rpc_plugin.cc

```diff
@@ -116,11 +116,22 @@ void SDLRPCPlugin::ProcessResumptionSubscription(
     SDLAppExtension& ext,
     resumption::Subscriber subscriber) {
   LOG4CXX_AUTO_TRACE(logger_);
-  application_manager::ApplicationSharedPtr application =
-      application_manager_->application(app.app_id());
+  LOG4CXX_DEBUG(
+      logger_,
+      "AKutsan app wants to subscribe for way points " << app.app_id());
   std::set<uint32_t> apps =
       application_manager_->GetAppsSubscribedForWayPoints();
-  application_manager_->SubscribeAppForWayPoints(application);
+  LOG4CXX_DEBUG(
+      logger_,
+      "AKutsan amount of apps already sybscribed for WP " << apps.size());
+  if (apps.size() > 0) {
```

Comment:
We already have function ApplicationManagerImpl::IsAnyAppSubscribedForWayPoints(), so it may be better to use it instead of GetAppsSubscribedForWayPoints() and (apps.size() > 0)

Reply:
But I still confused, why we need this check, because `function application_manager_->SubscribeAppForWayPoints` is not responsible for sending any messages to HMI, it only adds app to set with all subscribed apps and sends notification to mobile. As I understand, we need to use this function for all apps without any checks and handle situation with duplicate requests with `HandleResumptionSubscriptionRequest` method, isn't it?

Reply:
Fixed with 1701c0400 

We need to check this because if any application is already subscribed for waypoints we can just mark this one also as subscribed and skip resumption. 
HandleResumptionSubscriptionRequest will not care if app is subscribed to waypoints or not, it will just send appropriate request to HMI